src/atomate2/vasp/flows/defect.py

```diff
@@ -4,45 +4,217 @@
 from atomate2.vasp.sets.core import StaticSetGenerator
-from atomate2.vasp.sets.defect import AtomicRelaxSetGenerator
+from atomate2.vasp.sets.defect import (
+    SPECIAL_KPOINT,
+    ChargeStateRelaxSetGenerator,
+    ChargeStateStaticSetGenerator,
+    HSEChargeStateRelaxSetGenerator,
+)
 
 logger = logging.getLogger(__name__)
 
 
-DEFAULT_DISTORTIONS = (-1, -0.15, -0.1, -0.05, 0, 0.05, 0.1, 0.15, 1)
-DEFECT_INCAR_SETTINGS = {
-    "ISMEAR": 0,
-    "LWAVE": True,
-    "SIGMA": 0.05,
-    "KSPACING": None,
-    "ENCUT": 500,
-}
-DEFECT_KPOINT_SETTINGS = {"reciprocal_density": 64}
-
-DEFECT_RELAX_GENERATOR = AtomicRelaxSetGenerator(
-    use_structure_charge=True,
-    user_incar_settings=DEFECT_INCAR_SETTINGS,
-    user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
+"""
+Example Compound Maker to speed up defect relaxation calculations
+TODO: Update this into a faster maker:
+[PBE] -> (WAVECAR) -> [HSE]
+[PBE] -> (STRUCTURE) -> [HSE,gamma] -> (STRUCTURE) -> [HSE]
+"""
+
+HSE_DOUBLE_RELAX = DoubleRelaxMaker(
+    relax_maker1=RelaxMaker(
+        input_set_generator=ChargeStateRelaxSetGenerator(
+            user_kpoints_settings=SPECIAL_KPOINT
+        )
+    ),
+    relax_maker2=RelaxMaker(
+        input_set_generator=HSEChargeStateRelaxSetGenerator(
+            user_kpoints_settings=SPECIAL_KPOINT
+        ),
+        task_document_kwargs={"store_volumetric_data": ["locpot"]},
+        copy_vasp_kwargs={
+            "additional_vasp_files": ("WAVECAR",),
+            "contcar_to_poscar": True,
+        },
+    ),
 )
-DEFECT_STATIC_GENERATOR = StaticSetGenerator(
-    user_incar_settings=DEFECT_INCAR_SETTINGS,
-    user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
+
+DEFECT_STATIC_GENERATOR: StaticSetGenerator = StaticSetGenerator(
+    user_kpoints_settings=SPECIAL_KPOINT,
 )
 
+CCD_DEFAULT_DISTORTIONS = (-1, -0.15, -0.1, -0.05, 0, 0.05, 0.1, 0.15, 1)
+
+
+# Define RelaxMaker Errors
+def check_defect_relax_maker(maker: Maker):
+    """Check specific RelaxMaker settings.
+
+    Check all nested RelaxMaker in to make sure specific Setting is set.
+
+    Parameters
+    ----------
+    maker : Maker
+        The Maker object to check.
+    """
+    DUMMY_STRUCT = Structure(
+        Lattice.cubic(3.6), ["Si", "Si"], [[0.5, 0.5, 0.5], [0, 0, 0]]
+    )
+
+    def check_func(relax_maker: RelaxMaker):
+        input_gen = relax_maker.input_set_generator
+        vis = input_gen.get_input_set(DUMMY_STRUCT, potcar_spec=True)
+        if input_gen.use_structure_charge is False:
+            raise ValueError("use_structure_charge should be set to True")
+        if vis.incar["LVHAR"] is False:
+            raise ValueError("LVHAR should be set to True")
+        if vis.incar["ISIF"] != 2:
+            raise ValueError("ISIF should be 2")
+
+    recursive_call(maker, func=check_func, class_filter=RelaxMaker, nested=True)
+
+
+@dataclass
+class FormationEnergyMaker(Maker):
+    """Maker class to help calculate of the formation energy diagram.
+
```

Comment:
Can you add a detailed description of what this flow does, what the various options are etc.